fix: scramble for hex and dates

lukicenturi opened the pull request:
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

LefterisJP reviewed (commented):
Reviewed. Changes look good as a result but kraken is somehow affected as a name.


![2023-07-31_21-06](https://github.com/rotki/rotki/assets/1658405/1d96c02c-4e5f-47f9-a242-7aa06abff93c)

lukicenturi commented:
Updated

> Reviewed. Changes look good as a result but kraken is somehow affected as a name.
> 
> ![2023-07-31_21-06](https://user-images.githubusercontent.com/1658405/257322254-1d96c02c-4e5f-47f9-a242-7aa06abff93c.png)

Updated, this isn't affected by scramble setting anymore, changelog updated.

kelsos reviewed (approved):
Code looks good, since @LefterisJP tested that it works we can merged it :)
